src/components/post-form/post-form.tsx

```diff
@@ -0,0 +1,125 @@
+                        placeholder="cool post"
+                        value={data?.title}
+                    />
+                </label>
+                {errors.title && <p class={styles.error}>{errors.title}</p>}
+                <label>
+                    Tags (optional)
+                    <input
+                        type="text"
+                        name="tags"
+                        placeholder="comma, separated, with spaces"
+                        value={data?.tags.join(', ')}
+                    />
+                </label>
+                {errors.tags && <p class={styles.error}>{errors.tags}</p>}
+                <label>
+                    Body
+                    <textarea
+                        name="body"
+                        required
+                        minlength={1}
+                        placeholder="# hello, world!"
+                    >
+                        {data?.body}
+                    </textarea>
+                </label>
+                {errors.body && <p class={styles.error}>{errors.body}</p>}
+                <label>
+                    <input
+                        type="checkbox"
+                        name="unlisted"
+                        checked={data?.unlisted || true}
+                    />{' '}
+                    Unlisted (exclude this from search results)
+                </label>
+                {children}
+                <button class="success">Submit</button>
+            </section>
+        </form>
+    );
+};
+
+const NewPostForm = () => {
+    const [errors, setErrors] = useState<NewPostErrors>({});
+
+    async function submit(e: SubmitEvent) {
+        e.preventDefault();
+        setErrors({});
+        const formData = new FormData(e.target as HTMLFormElement);
+        const response = await fetch('/api/new', {
+            method: 'POST',
+            body: formData,
+        });
+        const data = await response.json();
+        if (response.status >= 400) {
+            setErrors(data.errors);
+        }
+        if (data.id) {
+            navigate(`/posts/${data.id}?created=${data.edit_key}`);
+        }
+    }
+    return <PostForm onSubmit={submit} errors={errors} />;
+};
+
+const EditPostForm = ({
+    postId,
+    editKey,
+    postData,
+}: {
+    postId: string;
+    editKey: string;
+    postData: PostData;
+}) => {
+    const [errors, setErrors] = useState<NewPostErrors>({});
+
+    async function submit(e: SubmitEvent) {
+        e.preventDefault();
+        setErrors({});
+        const formData = new FormData(e.target as HTMLFormElement);
+        const response = await fetch('/api/edit', {
+            method: 'POST',
+            body: formData,
+        });
+        const data = await response.json();
+        if (response.status >= 400) {
+            setErrors(data.errors);
+        }
+        if (data.id) {
+            navigate(`/posts/${data.id}?created=${data.edit_key}`);
+        }
+    }
+
+    return (
```

Comment:
Not entierly sure why, but when I submit an edit the whole form briefly reverts to its pre-edited state before going through.  It still works, but it's a bit disconcerting.

Reply:
I've noticed the same issue when submitting a post too, I think it has something to do with prefilling values on the server side. I'm not sure if there's an easy fix but I'll keep digging.